src/ui/patch_hub/patch_hub_core.sh

```diff
@@ -159,6 +159,16 @@ function show_registered_mailing_lists()
   message_box='Below, you can see the lore.kernel.org mailing lists that you have registered.'$'\n'
   message_box+='Select a mailing list to see the latest patchsets sent to it.'
 
+  fallback_message='kw could not find any registered mailing list.'$'\n'$'\n'
+  fallback_message+='Try to register mailing lists in the menu "Settings" then ' 
+  fallback_message+='"Register/Unregister Mailing Lists"'
+
+  if [[ "${#registered_mailing_lists[@]}" -eq 0 ]]; then
+      create_message_box 'Error' "${fallback_message}"
+      screen_sequence['SHOW_SCREEN']='dashboard'
+      return "$?"
+  fi
```

Comment:
For this whole `if` statement **block**, there are three things you should do:

- **Number 1**:

> There is a good chance that "$?" is 0, which means success, and from this code, I don't think you intend to return a success code. Am I correct? If so, you probably need to return 22 here or any other error code.

Actually, because of the FSM in `patch-hub`, returning a code 22 ends execution. Also, I feel that we can interpret that the "Error" is that there are no lists registered, and that this error is handled inside this `if` statement block, so we don't need to return an error code.

That being said, you should replace `return $?` by a simple `return`, like it is done across the feature for this handled "Errors".

- **Number 2**:

You used an indentation with four spaces in comparison with the `if` statement indentation, so you added one more indentation than needed. In other words,

```
if [[ "${#registered_mailing_lists[@]}" -eq 0 ]]; then
    create_message_box 'Error' "${fallback_message}"
    ...
```

should be

```
if [[ "${#registered_mailing_lists[@]}" -eq 0 ]]; then
  create_message_box 'Error' "${fallback_message}"
  ...
```

- **Number 3**:

Move this whole `if` statement block to between

```
  # Load registered mailing lists from config file into array
  IFS=',' read -r -a registered_mailing_lists <<< "${lore_config['lists']}"
```

and

```
  message_box='Below, you can see the lore.kernel.org mailing lists that you have registered.'$'\n'
  message_box+='Select a mailing list to see the latest patchsets sent to it.'
```

as, at this point, it is already possible to determine if there are no lists registered.